src/plugins/vis_default_editor/public/components/agg_params_helper.ts

```diff
@@ -93,7 +93,7 @@ function getAggParamsToRender({
         }
       }
       fields = filterAggTypeFields(availableFields, agg);
-      indexedFields = groupAndSortBy(fields, 'type', 'name');
+      indexedFields = groupAndSortBy(fields, 'type', 'displayName');
```

Comment:
if there were displayName  duplicates, you could prevent troubles with the EuiComboBox by appending the `name` in this case. So it could be `${displayName} (${name})`.

Reply:
Not sure if we really want this. I mean e.g. the shortDots setting will change the display name. If a user actually will use this, I assume they really want their names shortened, and I have the feeling printing the long name then for them again, kind of defines the purpose of why they used this setting?

Reply:
I'd just suggest to display it this way, if there are conflicts with duplicates. so users named 2 fields the same way.
or shortDots converts `i.have.cookies` and `i.hate.cookies` to `i.h.cookies`